scripts/dotnet-cli-build/PrepareTargets.cs

```diff
@@ -200,6 +172,162 @@ public static BuildTargetResult RestorePackages(BuildTargetContext c)
             return c.Success();
         }
 
+        [Target]
+        public static BuildTargetResult CheckPrereqDebianPackageBuildComponents(BuildTargetContext c)
+        {
+            if (!CurrentPlatform.IsUbuntu)
+            {
+                return c.Success();
+            }
+
+            var debianPackageBuildDependencies = new string[]
+            {
+                "devscripts", 
+                "debhelper",
+                "build-essential"
+            };
+
+            var messageBuilder = new StringBuilder();
+
+            foreach (var package in debianPackageBuildDependencies)
+            {
+                if (!AptPackageIsInstalled(package))
+                {
+                    messageBuilder.Append($"Error: Debian package build dependency {package} missing.");
+                    messageBuilder.Append(Environment.NewLine);
+                    messageBuilder.Append($"-> install with apt-get install {package}");
+                    messageBuilder.Append(Environment.NewLine);
+                }
+            }
+
+            if (messageBuilder.Length == 0)
+            {
+                return c.Success();
+            }
+            else
+            {
+                return c.Failed(messageBuilder.ToString());
+            }
+        }
+
+        [Target]
+        public static BuildTargetResult CheckPrereqCoreclrDependencyPackages(BuildTargetContext c)
+        {
+            if (!CurrentPlatform.IsUbuntu && !CurrentPlatform.IsCentOS)
```

Comment:
@brthor @anurse @livarcocc 

We need an approach for expressing conditional targets before we get too many of these. 

One approach would be to have explicit routers:
````
[Target]
public static BuildTargetResult CheckPrereqCoreclrDependencyPackages(BuildTargetContext c)
{
    switch(OS)
    {
        case Windows:
            c.RunTarget(CheckWindowsPrereqCoreclrDependencyPackages);
        ....
}
````

Can we do something cleaner? I'd really like to avoid targets that take on so many responsibilities...